### PR TITLE
fix(localdebug): use detect-port to detect if ports are in use

### PR DESCRIPTION
**Pull request by kuojianlu**
[detect-port](https://github.com/node-modules/detect-port) is used by react-scripts and many other projects.

To fix: https://msazure.visualstudio.com/Microsoft%20Teams%20Extensibility/_workitems/edit/13023066

### CODE REVIEW

File: packages/fx-core/package.json [hunk 1]
```diff
@@ -63,6 +63,7 @@
     "axios": "^0.21.2",
     "cryptr": "^6.0.2",
     "dateformat": "^4.5.1",
+    "detect-port": "^1.3.0",
```
Comment (swatDong) [inline packages/fx-core/package.json:66]: `detect-port` has its type definitions - https://www.npmjs.com/package/@types/detect-port

File: packages/fx-core/tests/common/local/portChecker.test.ts [hunk 2]
```diff
@@ -70,7 +70,7 @@ describe("portChecker", () => {
 
     it("happy path", async () => {
       const mockServer = new MockServer([]);
-      sinon.stub(net, "createServer").returns(mockServer as unknown as net.Server);
+      sinon.stub(net, "Server").returns(mockServer as unknown as net.Server);
```
Comment (swatDong) [inline packages/fx-core/tests/common/local/portChecker.test.ts:73]: Is it possible to directly mock `detect`?

File: packages/fx-core/package.json (hunk 1 above)
Comment (kuojianlu) [reply packages/fx-core/package.json:66]: Good catch, updated.

File: packages/fx-core/tests/common/local/portChecker.test.ts (hunk 2 above)
Comment (kuojianlu) [reply packages/fx-core/tests/common/local/portChecker.test.ts]: Yes, mocked `detect-port` instead.

File: packages/fx-core/package.json [hunk 3]
```diff
@@ -84,6 +85,7 @@
     "node-ts-uuid": "^1.0.8",
     "openapi-types": "^7.2.3",
     "proper-lockfile": "^4.1.2",
+    "proxyquire": "^2.1.3",
```
Comment (swatDong) [inline packages/fx-core/package.json:88]: Move to `devDependencies` if it's only used by test.

**Review by swatDong — APPROVED** [review]

**Pull request merged**